### PR TITLE
[feat] Add the ability to use automatic subscriber assignment to the subscriber settings.

### PR DESCRIPTION
**Pull request by dpcollins-google**
Diffbased on #135

DO_NOT_MERGE

### CODE REVIEW

File: google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/AssignerBuilder.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        try {
+          stub =
+              Stubs.defaultStub(
+                  Endpoints.regionalEndpoint(
+                      SubscriptionPaths.getZone(builder.subscriptionPath()).region()),
+                  PartitionAssignmentServiceGrpc::newStub);
+        } catch (IOException e) {
+          throw Status.INTERNAL
+              .withCause(e)
+              .withDescription("Creating cursor stub failed.")
```
Comment (manuelmenzella-google) [inline google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/AssignerBuilder.java:79]: s/cursor/assigner/g.

File: google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/AssignerBuilder.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      uuidBuffer.putLong(uuid.getMostSignificantBits());
+      uuidBuffer.putLong(uuid.getLeastSignificantBits());
+
+      InitialPartitionAssignmentRequest initial =
+          InitialPartitionAssignmentRequest.newBuilder()
+              .setSubscription(builder.subscriptionPath().value())
+              .setClientId(ByteString.copyFrom(uuidBuffer))
+              .build();
+      return new AssignerImpl(
+          stub, new ConnectedAssignerImpl.Factory(), initial, builder.receiver());
```
Comment (manuelmenzella-google) [inline google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/AssignerBuilder.java:95]: Nit: check that receiver is actually populated?

Comment (dpcollins-google) [reply google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/AssignerBuilder.java:95]: Autovalue does this already. its not optional.

Comment (dpcollins-google) [reply google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/AssignerBuilder.java:95]: Autovalue does this already. Its not optional.

File: google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/AssignerBuilder.java (hunk 1 above)
Comment (dpcollins-google) [reply google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/AssignerBuilder.java]: Done.

**Comment by dpcollins-google** [comment]
Something is broken with github and it isn't recognising new changes. Recreating

**Pull request closed without merging**